### PR TITLE
updated correct API to get index templates etc

### PR DESCRIPTION
**Pull request by ramdaspotale**
Hi, I think documentation needs correction on below points.
1. updated correct API to get index template
2.  on line 18, it should be index pattern instead of template pattern
3.  index pattern should have es version as 8 which matches elasticsearch version, also updated index patterns in API request payload

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by ramdaspotale** [comment]
pinging @elastic/es-docs

**Comment by ramdaspotale** [comment]
@elasticsearchmachine

**Comment by abdonpijpelink** [comment]
@ramdaspotale thank you for updating the docs! Your PR is targeting 8.6. To ensure this is also updated in future documentation, the PR should target the `main` branch instead. Can you rebase it, or would you like me to do it for you?

**Comment by ramdaspotale** [comment]
@abdonpijpelink thanks for your update..i am okay if you can rebase it to main

File: docs/reference/monitoring/indices.asciidoc [hunk 1]
```diff
@@ -28,7 +28,7 @@ and the number of replicas to two.
 ----------------------------------
 PUT /_template/custom_monitoring
```
Comment (abdonpijpelink) [inline docs/reference/monitoring/indices.asciidoc:29]: We should also change this request:
- use the `_index_template` endpoint 
- `order` is now `priority` (need to change it in the text above, on line 19, too)
- the `settings` object needs to be wrapped in a `template` object

File: docs/reference/monitoring/indices.asciidoc [hunk 2]
```diff
@@ -9,13 +9,13 @@ You can retrieve the templates through the `_template` API:
...
 ----------------------------------
-GET /_template/.monitoring-*
+GET /_index_template/.monitoring-es-mb*
 ----------------------------------
 
 By default, the template configures one shard and one replica for the
 monitoring indices. To override the default settings, add your own template:
 
-. Set the `template` patterns to match existing `.monitoring-{product}-7-*` indices.
+. Set the `index` patterns to match existing `.monitoring-{product}-8-*` indices.
```
Comment (abdonpijpelink) [inline docs/reference/monitoring/indices.asciidoc:18]:
```suggestion
. Set `index_patterns` to match existing `.monitoring-{product}-8-*` indices.
```

**Review by abdonpijpelink — CHANGES_REQUESTED** [review]
This looks good. I've left some suggestions.

Also, on line 5, we should link to docs for the new index templates (change `<<indices-templates-v1` into `<<index-templates`)

File: docs/reference/monitoring/indices.asciidoc (hunk 1 above)
Comment (ramdaspotale) [reply docs/reference/monitoring/indices.asciidoc]: I have updated the doc as per your suggestions. please verify

Comment (abdonpijpelink) [reply docs/reference/monitoring/indices.asciidoc]: `settings` needs to be wrapped in a `template` object. The request becomes:

```
PUT /_index_template/custom_monitoring
{
  "index_patterns": [
    ".monitoring-beats-8-*",
    ".monitoring-es-8-*",
    ".monitoring-kibana-8-*",
    ".monitoring-logstash-8-*"
  ],
  "priority": 1,
  "template": {
    "settings": {
      "number_of_shards": 5,
      "number_of_replicas": 2
    }
  }
}
```

**Review by ramdaspotale — COMMENTED** [review]
changes done as requested.

**Comment by ramdaspotale** [comment]
apologies..i missed that...i will add settings to template object.

**Review by ramdaspotale — COMMENTED** [review]
`settings` added inside template object.

File: docs/reference/monitoring/indices.asciidoc (hunk 1 above)
Comment (ramdaspotale) [reply docs/reference/monitoring/indices.asciidoc]: please verify now

**Review by abdonpijpelink — APPROVED** [review]
LGTM. Thank you @ramdaspotale !

**Comment by abdonpijpelink** [comment]
@elasticmachine test this please

**Comment by ramdaspotale** [comment]
@abdonpijpelink  looks like builds finished

**Pull request merged**